libs/vertexai/langchain_google_vertexai/functions_utils.py

```diff
@@ -94,6 +94,31 @@ class Config:
         extra = "allow"
 
 
+def _replace_all_ofs(schema: Dict[str, Any]) -> Dict[str, Any]:
+    """Assumes dereferenced schema."""
+    new_schema: Dict[str, Any] = {}
+    for k, v in schema.items():
+        if isinstance(v, dict) and "allOf" in v:
+            if isinstance(v["allOf"], list) and len(v["allOf"]) == 1:
+                obj = dict(v["allOf"][0])
+                obj["title"] = v.get("title", obj.get("title", ""))
+                obj["description"] = " ".join(
+                    (v.get("description", ""), obj.get("description", ""))
```

Comment:
I'm not sure about joining descriptions.

This code:
```python
class Node(BaseModel):
    """This is a node"""
    id: str
    type: str

class Relationship(BaseModel):
    source: Node
    target: Node = Field(..., description="This is a target node")
```
will produce following definition for target property:
```json
{
  "target": {
    "title": "Target",
    "description": "This is a target node This is a node",
    "type": "object",
    "properties": {
      "id": {
        "title": "Id",
        "type": "string"
      },
      "type": {
        "title": "Type",
        "type": "string"
      }
    },
    "required": ["id", "type"]
  }
}
```

Reply:
yea not ideal but is there a better generic solution? both descriptions could have important (and distinct) info

Reply:
I would consider coalescing i. e. field description should take precedence

```
class Relationship(BaseModel):
    source: Node
    target: Node = Field(..., description="This is a target node")
```

over model description:
```
class Node(BaseModel):
    """This is a node"""
    id: str
    type: str
```

And could you please describe reasoning and behaviour in the function doc string?

Reply:
@lkuligin what is your take on that?